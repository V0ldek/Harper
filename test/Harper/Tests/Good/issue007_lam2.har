main :: Integer;
main = {
  var (x :: Integer) = 42;
  var (y :: Integer) = 3;
  var (z :: Integer) = 17;
  var (a :: Integer) = 7;
  (f :: Integer -> Integer -> Integer -> Integer) = 
    \(b :: Integer) (c :: Integer) (x :: Integer) => a * b + c - x + y * z;
  x += 1;
  y -= 50;
  z /= 4;
  a ^= 2;
  return f 1 2 3;
};