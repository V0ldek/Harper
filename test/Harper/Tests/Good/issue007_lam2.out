Execution ended with value: -140